tools/mkcomposefs.c

```diff
@@ -893,6 +876,307 @@ static ssize_t write_cb(void *_file, void *buf, size_t count)
 	return fwrite(buf, 1, count, file);
 }
 
+static void log_message_with_time( const char* msg )
+{
+	time_t t 		= time(NULL);
+  	struct tm tm 	= *localtime(&t);
+	fprintf(stdout, "%02d:%02d:%02d\t %s\n", 
+				tm.tm_hour, tm.tm_min, tm.tm_sec, msg);
+}
+
+struct ParallelCopyInfo
+{
+	struct lcfs_node_s		*node;
+	struct ParallelCopyInfo	*next;
+	char 					*parentPath;
+};
+
+static struct ParallelCopyInfo* new_parallel_copy_info(void)
+{
+	struct ParallelCopyInfo *ret = 
+		calloc(1, sizeof( struct ParallelCopyInfo));
+	ret->next 		= NULL;
```

Comment:
calloc already NULLed these.

Reply:
Changed this to an array instead of linked list. This code is not valid anymore.